spec/features/workflow/edit_backdate_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+RSpec.feature "Edit backdate" do
+  scenario do
+    given_there_is_an_editable_backdated_edition
+    when_i_visit_the_summary_page
+    and_i_click_to_edit_the_backdate
+    and_i_enter_a_new_backdate_and_save
+    then_i_see_timeline_entries_for_the_old_and_new_backdates
+  end
+
+  def given_there_is_an_editable_backdated_edition
+    @old_backdate = Time.current.yesterday
+    @edition = create(:edition, backdated_to: @old_backdate)
+  end
+
+  def when_i_visit_the_summary_page
+    visit document_path(@edition.document)
+  end
+
+  def and_i_click_to_edit_the_backdate
+    click_on "Edit Backdate"
+  end
+
+  def and_i_enter_a_new_backdate_and_save
+    stub_publishing_api_put_content(@edition.content_id, {})
+
+    fill_in "backdate[date][day]", with: "1"
+    fill_in "backdate[date][month]", with: "1"
+    fill_in "backdate[date][year]", with: "2019"
+    click_on "Save"
+  end
+
+  def then_i_see_timeline_entries_for_the_old_and_new_backdates
+    formatted_old_backdate = @old_backdate.strftime("%-d %B %Y")
+
+    expect(page).to have_content(
+      I18n.t!("documents.history.entry_types.backdated", date: formatted_old_backdate),
+    )
+    expect(page).to have_content(
+      I18n.t!("documents.history.entry_types.backdated", date: "1st January 2019"),
+    )
+  end
+end
```

Comment:
I'm a bit confused here as to how this test works. Since we created the backdated edition using FactoryBot and set to a backdate I don't know how generated the timeline entry that we confirm exists in line 38 :-/

Regardless of that I'm wondering whether we should have a test for this, mostly as it's whole extra feature test for a small tweak. I can definitely see why you put it together as there doesn't seem to be a clear way to test this exact scenario. I'd be satisfied though if you just didn't have a specific test for it (it's already covered in the general case in other tests so it has coverage)

Reply:
Fair enough, I thought that since it was a bug fix we should add a test to catch it, but agree it's not adding a tonne of value (and now you mention it, it is surprising that the timeline entry we check for in line 38 is there!).